contracts/facets/EmbalmerFacet.sol

```diff
@@ -0,0 +1,130 @@
+        require(
+            !s.sarcophaguses[identifier].exists,
+            "sarcophagus already exists"
+        );
+
+        // Confirm that the ressurection time is in the future
+        require(
+            resurrectionTime > block.timestamp,
+            "resurrection time must be in the future"
+        );
+
+        // Initialize a list of archaeologist addresses
+        address[] memory archaeologistAddresses = new address[](
+            archaeologists.length
+        );
+
+        for (uint256 i = 0; i < archaeologists.length; i++) {
+            // Calculate the amount of cursed bond the archaeologists needs to lock up
+            uint256 cursedBondAmount = LibBonds.calculateCursedBond(
+                archaeologists[i].diggingFee,
+                archaeologists[i].bounty
+            );
+
+            // Confirm that the archaeologist has enough free bond.
+            // This error could mean that the archaeologist has either run out
+            // of free bond or has never even interacted with sarcophagus
+            require(
+                s.freeBonds[archaeologists[i].archAddress] >= cursedBondAmount,
+                "archaeologist does not have enough free bond"
+            );
+
+            // Lock up the archaeologist's bond by the cursed bond amount
+            LibBonds.lockUpBond(
+                archaeologists[i].archAddress,
+                cursedBondAmount
+            );
+
+            // Add to the neccessary data structures
+            s.archaeologistSarcophaguses[archaeologists[i].archAddress].push(
+                identifier
+            );
+            s.sarcophagusArchaeologists[identifier][
+                archaeologists[i].archAddress
+            ] = archaeologists[i];
+
+            // Add the archaeologist address to the list of addresses to be
+            // passed in to the sarcophagus object
+            archaeologistAddresses[i] = archaeologists[i].archAddress;
+        }
+
+        s.sarcophaguses[identifier] = LibTypes.Sarcophagus({
+            name: name,
+            exists: true,
+            canBeTransferred: canBeTransfered,
+            resurrectionTime: resurrectionTime,
+            arweaveTxId: "",
+            embalmer: msg.sender,
+            recipientAddress: recipient,
+            arweaveArchaeologist: arweaveArchaeologist,
+            archaeologists: archaeologistAddresses
+        });
+
+        // Add the identifier to necessary data structures
+        s.sarcophagusIdentifiers.push(identifier);
+        s.embalmerSarcophaguses[msg.sender].push(identifier);
+        s.recipientSarcophaguses[recipient].push(identifier);
+
+        // Calculate the total fees in sarco tokens that the contract will
+        // receive from the embalmer
+        uint256 totalFees = LibBonds.calculateTotalFees(
+            archaeologists,
+            s
+            .sarcophagusArchaeologists[identifier][arweaveArchaeologist]
+                .storageFee
```

Comment:
Does the contract need to do any validation around this `storageFee` amount? Maybe it doesn't, just wondering if there would be issues if it was undefined.

Reply:
Is there a situation where a `uint256` can be undefined? It should only ever be 0 or a positive number, so I think that no validation would be necessary.